conda/common/configuration.py

```diff
@@ -1600,3 +1601,56 @@ def typify_parameter(self, parameter_name, value, source):
 
     def get_descriptions(self):
         raise NotImplementedError()
+
+
+def unique_sequence_map(
+    *, unique_key: str, property_name: str, allowed_keys: set | None = None
+):
+    """
+    Used to validate properties on :class:`Configuration` subclasses defined as a
+    ``SequenceParameter(MapParameter())`` where the map contains a single key that
+    should be regarded as unique. This decorator will handle removing duplicates and
+    merging to a single sequence.
+    """
+
+    def inner_wrap(func):
+        @wraps(func)
+        def wrapper(self, *args, **kwargs):
+            sequence_map = func(self, *args, **kwargs)
+            known_keys = set()
+            new_sequence_mapping = []
```

Comment:
We should use a `dict` instead with the `unique_key` as they key:

```suggestion
            new_sequence_mapping = {}
```

Reply:
@kenodegard,

I don't understand why this should be dictionary instead of a list. Could you please explain why?

Reply:
The `known_keys` and `new_sequence_mapping` are tightly coupled with a one-to-one relationship:

```
        def wrapper(*args, **kwargs):
            new_sequence_mapping = {}

            for mapping in func(*args, **kwargs):
                unique_key_value = mapping.get(unique_key)

                if unique_key_value is None:
                    ...

                if unique_key_value in known_keys:
                    ...

                if allowed_keys is not None and unique_key_value not in allowed_keys:
                    ...

                new_sequence_mapping[unique_key_value] = mapping

            return tuple(new_sequence_mapping.values())
```